### DISCUSSION TITLE
Sharing Album error

### DISCUSSION THREAD

**Discussion by Wingstyle** [photoprism/photoprism#2307] [Q&A]
I am trying to share an album but can't get it to work. I click to get the link in my clipboard and paste into my browser. When I do I get: 
Unable to connect

Firefox can’t establish a connection to the server at localhost:2342.

    The site could be temporarily unavailable or too busy. Try again in a few moments.
    If you are unable to load any pages, check your computer’s network connection.
    If your computer or network is protected by a firewall or proxy, make sure that Firefox is permitted to access the Web.

I also tried Chrome with the same results. Does anyone have any suggestions? 

By the way, I haven't been using PhotoPrism very long, but I like it so far. Looking forward to the hierarchical folder addition so many Synology people are asking for. Thanks for all your hard work.

**Comment by lastzero** [comment]
It's the default URL, if you are using a different host name, please update the site URL in the configuration (docker-compose.yml) and restart.

**Comment by Wingstyle** [comment]
Thanks for your quick reply. 

So you are saying to change the default localhost:2342 to my NAS url:port? That makes sense. Where do I find docker-compose.yml? 

I am pretty new to this, but with help on another forum managed to move photoprism/storage to a share folder on my nas, and using portainer, set bind volume with photoprism/storage to the share folder on my nas. Also set a bind volume with photoprism/originals to the share folder on my nas that contains my pictures. Everything seems to be working. I looked in photoprism/storage and see a lot of yml files, but not docker-compose.

**Reply by lastzero** [reply]
Portainer is your docker-compose.yml in this case, so not a config file but a config web UI. You should be able to set service variables somewhere in the Portainer UI. Set PHOTOPRISM_SITE_URL to the name or IP Adress of your server including http:// or https://.

**Comment by Wingstyle** [comment]
Thanks again Lastzero for the quick reply.

I missed PHOTOPRISM_SITE_URL when I was looking in Portainer before posting here for help. Thanks for giving me the exact thing to look for. I made the change and it almost fixed it. I now get a Photoprism page with the your logo in the upper left of the page, and the "PhotoPrism needs your support" link. Nothing else, and no way to do anything. I really don't think this is a Photoprism issue, but any suggestions would be appreciated. 

I am going through a reverse proxy on my Asustor NAS. When I test the link while setting up the reverse proxy I get to the photoprism page. But when I use the reverse proxy url, I get the same page as with the shared folder link photoprism generates.

**Comment by jab8778** [comment]
I just successfully installed PhotoPrism on Asustor AS-5202T. I set up a reverse proxy. I am a newbee to Docker/Portainer, but after a few hours of trial and error, I was able to get it done.
Before installing PhotoPrism there are a few apps that need to be installed from Asustor App Central:
Apache HTTP Server
Docker Engine
FFmpeg
MariaDB
php8
Portainer CE

1-Under the General Tab make note of the port numbers you have for HTTP and HTTPS
2-Go to your Routers Port Forwarding Page and add the two ports from 1. 
3-Also add the two ports for PhotoPrism (default are 2342 and 32770).
4-That should be all you need to do on the Router, now back to your NAS Settings page.
5-Go to Manual Connect and enable DDNS Service.
6-Select a service - easiest would be myasustor.com.
7-The email and password should populate from your Asustor ID.
8-Create a Cloud ID, hit Refresh to make sure it works, then Apply.
9-Got to the next tab for EZ-Router, if the ports you forwarded on your router show as a red x or a green check, it doesn't seem to matter. I have seen it both ways, the way to verify your forwarding works is to go to your phone, turn off wi-fi, then type in ClouID.myasustor.com:8001 (or whatever port you have set in 1 for HTTPS.) You should be directed to your NAS login screen.
10. Now it's time to set up a Reverse Proxy in the NAS under Services - Reverse Proxy.
11. Proxy Domain Page
     a. Title - Anything
     b. Protocol - HTTPS
     c. Domain Name - Your CloudID.myasustor.com
     d. Port number - 2342 
     e. Network Interface - *
     f. Path - /
     g. Check the box to Enable EZ-Connect port forwarding
12. Reverse Proxy Domain 
     a. Name - Anything
     b. Protocol - HTTP
     c.  Host Name - The numeric IP Address of your NAS XXX.XXX.XX.XX
     d. Port - 32770
     e. Path - /
     f. Test connection - you should get the same result you did in your original post. Now you need to edit the PhotoPrism Container in Portainer CE.
13. Log into Portainer CE.
14. On the Hoe Screen click on the Docker logo.
15. On the Dashboard click on Containers.
16. On the container list find and click on the hyperlink for PhotoPrismD.
17. Under Actions, click Stop.
18. Under Actions, click Duplicate/Edit.
19. Scroll down to Advanced Container Settings.
20. You should see ENV, click on it.
21. Find the entry PHOTOPRISM_SITE_URL, it should have http://photoprism.me:2342/, you need to replace that with your CloudID - https://CLOUDID.myasustor.com:2342/.
22. Scroll up to Actions - click on Deploy the Container
23.  I went back to the top and clicked on Restart then logged out.
24. Now log on to PhotoPrism, select an album to share, click on the drop down for the share link and it should now begin with https://CLOUDID.myasustor.com:2342/followed by the share link.
25. Copy the link and try it on a device that is not on your local network.
Enjoy!
